src/rust/engine/options/src/lib.rs

```diff
@@ -158,27 +159,62 @@ pub struct OptionParser {
 
 impl OptionParser {
     pub fn new(env: Env, args: Args) -> Result<OptionParser, String> {
+        let buildroot = BuildRoot::find()?;
+        let buildroot_string = String::from_utf8(buildroot.as_os_str().as_bytes().to_vec())
+            .map_err(|e| {
+                format!(
+                    "Failed to decode build root path {}: {}",
+                    buildroot.display(),
+                    e
+                )
```

Comment:
I realised that this is just moved code, but I think this could probably call https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.to_str, although that loses the precise details of the error:

```suggestion
        let buildroot_string = buildroot.to_str()
            .ok_or_else(|e| {
                format!(
                    "Failed to decode build root path {} as UTF-8",
                    buildroot.display(),
                )
```

Reply:
Yeah, I think John had it this way for a reason, so we can see more detail.